src/core/meson.build

```diff
@@ -131,9 +131,25 @@ libcore_sources = '''
         unit.h
 '''.split()
 
+libcore_dependencies = [
+        versiondep,
+        threads,
+        libdl,
+        librt,
+        libseccomp,
+        libpam,
+        libaudit,
+        libkmod,
+        libapparmor,
+        libselinux,
+        libmount,
+        libacl
+]
+
 subdir('bpf/socket_bind')
 if conf.get('BPF_FRAMEWORK') == 1
         libcore_sources += [socket_bind_skel_h]
+        libcore_dependencies += [libbpf]
```

Comment:
We dlopen the library, won't this cause it to link instead?

Reply:
What issue is there with linking to the library?

Reply:
It's one of many optionals used via dlopen